extmod/modlwip.c

```diff
@@ -1168,16 +1170,17 @@ STATIC mp_uint_t lwip_socket_ioctl(mp_obj_t self_in, mp_uint_t request, uintptr_
             ret |= MP_STREAM_POLL_RD;
         }
 
-        if (flags & MP_STREAM_POLL_WR && tcp_sndbuf(socket->pcb.tcp) > 0) {
+        if (flags & MP_STREAM_POLL_WR && (socket->pcb.tcp == NULL || tcp_sndbuf(socket->pcb.tcp) > 0)) {
```

Comment:
Shouldn't this be `(socket->pcb.tcp != NULL && ...`?  Is it still writable if the pcb is NULL?

Reply:
On posix systems, when a socket is closed, it is being writable, where a write operation will return an error.